src/test/java/com/google/sps/MutationTest.java

```diff
@@ -396,6 +406,80 @@ public void addTokens() {
     Assert.assertEquals(newNodeA.tokenList(), newTokens);
   }
 
+  /*
+   * Added to address a bug in graph generation where node instance in
+   * graph and map were different therefore mutating one didn't affect
+   * the other. Originally, node B was first added to both the graph
+   * and map as a child of A but only updated in the map when it itself
+   * was processed. This test checks that the two are the same.
+   */
+  @Test
+  public void addTokensToChild() {
```

Comment:
Is this a bug that the current PR fixed? I didn't see any related code changes for it?

Reply:
ohh I think this was merged from another branch (client). I believe it was the bug that 
`      if (!graphNodesMap.containsKey(nodeName)) {
        roots.add(nodeName);
        graph.addNode(graphNode);
        graphNodesMap.put(nodeName, graphNode);
      }` solved in DataServlet! (sorry this was confusing, probably would've been good to put this test on the same branch as that code)